File: frogment/src/main/java/ch/grze/frogment/core/callbacks/ActivityComponentInjectorCallbacks.java
package ch.grze.frogment.core.callbacks;

import android.app.Activity;
import android.os.Bundle;

import ch.grze.frogment.core.Core;

public class ActivityComponentInjectorCallbacks extends AbstractActivityLifecycleCallbacks {
    public ActivityComponentInjectorCallbacks(Core core) {
        super(core);
    }

    @Override
    public void onActivityCreated(Activity activity, Bundle bundle) {
        super.onActivityCreated(activity, bundle);

        core.injectComponents(activity);
    }
}

File: frogment/src/main/java/ch/grze/frogment/core/callbacks/ActivityComponentInjectorCallbacks.kt
package ch.grze.frogment.core.callbacks

import android.app.Activity
import android.os.Bundle

import ch.grze.frogment.core.Core

class ActivityComponentInjectorCallbacks(core: Core) : AbstractActivityLifecycleCallbacks(core) {
    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
        super.onActivityCreated(activity, bundle)

        core.injectComponents(activity)
    }
}
